consensus/src/chained_bft/event_processor.rs

```diff
@@ -728,6 +728,15 @@ impl<T: Payload> EventProcessor<T> {
         let deadline = self.pacemaker.current_round_deadline();
         let preferred_peer = vote.author();
         // TODO [Reconfiguration] Verify epoch of the vote message.
+        if self.pacemaker.current_round() != vote.block_round() {
```

Comment:
We should probably need to think more about this condition. 
Consider a scenario, in which a proposal at round `r` first arrives to `v1`, which votes for this proposal and sends the vote message to `v2`, which didn't receive the original proposal yet. In this case we want `v2` to aggregate the vote but it's still at round `r-1`.
The vote messages today don't carry `SyncInfo`, hence `v2` cannot move from round `r-1` to round `r` merely due to the certificates carried by the vote message. I think we could add this check only after we have `SyncInfo` carried by the vote messages.

Note that the fix in the vote aggregation doesn't depend on this check, so removing this check is perfectly fine in this change.

Reply:
Ohh yes, we talked about SyncInfo but forgot to remove the check. Fixed.